build: use the system provided LuaJIT on s390x

rightblank opened the pull request:
<!-- Provide summary of changes -->

Hi, @edsiper, this PR follows #7286 to allow to use system provided LuaJIT on **s390x**, please help to review and merge it, thanks!

<!-- Issue number, if available. E.g. "Fixes #31", "Addresses #42, #77" -->

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [x] Example configuration file for the change
<details><summary>fluent.conf</summary>

```
[INPUT]
    name                tail
    alias               input_tail_logs
    tag                 some_tag
    buffer_chunk_size   5M
    buffer_max_size     5M
    db                  /tmp/fluentbit-lua-test.db
    mem_buf_limit       5M
    path                fake.log
    read_from_head      true
    refresh_interval    5s
    skip_empty_lines    on
    skip_long_lines     on


[FILTER]
    name            lua
    alias           filter_lua_logs
    match           some_tag
    call            ModifyRecord
    protected_mode  true
    script          test.lua
    time_as_table   true

[OUTPUT]
    name   stdout
    match  some_tag
```

</details> 

<details><summary>test.lua</summary>

```lua
function ModifyRecord(tag, timestamp, record)
    print("[in_lua]: tag: " .. tag .. "timestamp: ", timestamp, "log: " .. record["log"] )
    return 0, timestamp, record
end
``` 

</details> 

<details><summary>fake.log</summary>

```
Aug  5 03:06:06 fluentbitvm sshd[935]: Server listening on 0.0.0.0 port 22.
Aug  5 03:06:06 fluentbitvm sshd[935]: Server listening on :: port 22.
Aug  5 03:08:34 fluentbitvm sshd[1060]: Accepted publickey for linux1 from MASKED_IP port 22898 ssh2: ED25519 SHA256:K89pcH2o9cPK0hvAr4LT+BtI7/xi6xGERu48CHsSMKA
Aug  5 03:08:34 fluentbitvm sshd[1060]: pam_unix(sshd:session): session opened for user linux1(uid=1000) by (uid=0)
Aug  5 03:08:34 fluentbitvm systemd-logind[865]: New session 1 of user linux1.
Aug  5 03:08:34 fluentbitvm systemd: pam_unix(systemd-user:session): session opened for user linux1(uid=1000) by (uid=0)
Aug  5 03:10:01 fluentbitvm CRON[1170]: pam_unix(cron:session): session opened for user root(uid=0) by (uid=0)
Aug  5 03:10:01 fluentbitvm CRON[1170]: pam_unix(cron:session): session closed for user root
Aug  5 03:10:03 fluentbitvm sshd[1172]: Accepted publickey for linux1 from MASKED_IP port 22921 ssh2: ED25519 SHA256:K89pcH2o9cPK0hvAr4LT+BtI7/xi6xGERu48CHsSMKA
Aug  5 03:10:03 fluentbitvm sshd[1172]: pam_unix(sshd:session): session opened for user linux1(uid=1000) by (uid=0)
```

</details>

 
- [x] Debug log output from testing the change
<details><summary>Debug output</summary>

```
linux1@fluentbitvm:~/code/fluent-bit/9172_debug/bin$  ./fluent-bit -vvv -c fluent.conf 
Fluent Bit v3.1.5
* Copyright (C) 2015-2024 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

______ _                  _    ______ _ _           _____  __  
|  ___| |                | |   | ___ (_) |         |____ |/  | 
| |_  | |_   _  ___ _ __ | |_  | |_/ /_| |_  __   __   / /`| | 
|  _| | | | | |/ _ \ '_ \| __| | ___ \ | __| \ \ / /   \ \ | | 
| |   | | |_| |  __/ | | | |_  | |_/ / | |_   \ V /.___/ /_| |_
\_|   |_|\__,_|\___|_| |_|\__| \____/|_|\__|   \_/ \____(_)___/

[2024/08/08 07:07:02] [ info] Configuration:
[2024/08/08 07:07:02] [ info]  flush time     | 1.000000 seconds
[2024/08/08 07:07:02] [ info]  grace          | 5 seconds
[2024/08/08 07:07:02] [ info]  daemon         | 0
[2024/08/08 07:07:02] [ info] ___________
[2024/08/08 07:07:02] [ info]  inputs:
[2024/08/08 07:07:02] [ info]      tail
[2024/08/08 07:07:02] [ info] ___________
[2024/08/08 07:07:02] [ info]  filters:
[2024/08/08 07:07:02] [ info]      lua.0
[2024/08/08 07:07:02] [ info] ___________
[2024/08/08 07:07:02] [ info]  outputs:
[2024/08/08 07:07:02] [ info]      stdout.0
[2024/08/08 07:07:02] [ info] ___________
[2024/08/08 07:07:02] [ info]  collectors:
[2024/08/08 07:07:02] [ info] [fluent bit] version=3.1.5, commit=52ec818316, pid=276536
[2024/08/08 07:07:02] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2024/08/08 07:07:02] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/08 07:07:02] [ info] [cmetrics] version=0.9.3
[2024/08/08 07:07:02] [ info] [ctraces ] version=0.5.3
[2024/08/08 07:07:02] [ info] [input:tail:input_tail_logs] initializing
[2024/08/08 07:07:02] [ info] [input:tail:input_tail_logs] storage_strategy='memory' (memory only)
[2024/08/08 07:07:02] [debug] [tail:input_tail_logs] created event channels: read=21 write=22
[2024/08/08 07:07:02] [debug] [input:tail:input_tail_logs] flb_tail_fs_inotify_init() initializing inotify tail input
[2024/08/08 07:07:02] [debug] [input:tail:input_tail_logs] inotify watch fd=30
[2024/08/08 07:07:02] [debug] [input:tail:input_tail_logs] scanning path fake.log
[2024/08/08 07:07:02] [debug] [input:tail:input_tail_logs] file will be read in POSIX_FADV_DONTNEED mode fake.log
[2024/08/08 07:07:02] [debug] [input:tail:input_tail_logs] inode=548098 with offset=1115 appended as fake.log
[2024/08/08 07:07:02] [debug] [input:tail:input_tail_logs] scan_glob add(): fake.log, inode 548098
[2024/08/08 07:07:02] [debug] [input:tail:input_tail_logs] 1 new files found on path 'fake.log'
[2024/08/08 07:07:02] [ info] [input:tail:input_tail_logs] db: delete unmonitored stale inodes from the database: count=0
[2024/08/08 07:07:02] [debug] [stdout:stdout.0] created event channels: read=32 write=33
[2024/08/08 07:07:02] [ info] [sp] stream processor started
[2024/08/08 07:07:02] [ info] [output:stdout:stdout.0] worker #0 started
[2024/08/08 07:07:02] [debug] [input:tail:input_tail_logs] inode=548098 file=fake.log promote to TAIL_EVENT
[2024/08/08 07:07:02] [ info] [input:tail:input_tail_logs] inotify_fs_add(): inode=548098 watch_fd=1 name=fake.log
[2024/08/08 07:07:02] [debug] [input:tail:input_tail_logs] [static files] processed 0b, done
[2024/08/08 07:07:07] [debug] [input:tail:input_tail_logs] scanning path fake.log
[2024/08/08 07:07:07] [debug] [input:tail:input_tail_logs] scan_blog add(): dismissed: fake.log, inode 548098
[2024/08/08 07:07:07] [debug] [input:tail:input_tail_logs] 0 new files found on path 'fake.log'
^C[2024/08/08 07:07:07] [engine] caught signal (SIGINT)
[2024/08/08 07:07:07] [trace] [engine] flush enqueued data
[2024/08/08 07:07:07] [ warn] [engine] service will shutdown in max 5 seconds
[2024/08/08 07:07:07] [ info] [input] pausing input_tail_logs
[2024/08/08 07:07:08] [ info] [engine] service has stopped (0 pending tasks)
[2024/08/08 07:07:08] [ info] [input] pausing input_tail_logs
[2024/08/08 07:07:08] [ info] [output:stdout:stdout.0] thread worker #0 stopping...
[2024/08/08 07:07:08] [ info] [output:stdout:stdout.0] thread worker #0 stopped
[2024/08/08 07:07:08] [debug] [input:tail:input_tail_logs] inode=548098 removing file name fake.log
[2024/08/08 07:07:08] [ info] [input:tail:input_tail_logs] inotify_fs_remove(): inode=548098 watch_fd=1

```

</details> 
<!--
Please refer to the Developer Guide for instructions on building Fluent Bit with Valgrind support:
https://github.com/fluent/fluent-bit/blob/master/DEVELOPER_GUIDE.md#valgrind
Invoke Fluent Bit and Valgrind as: $ valgrind --leak-check=full ./bin/fluent-bit <args>
-->

- [x] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

<details><summary>Valgrind output</summary>

```
linux1@fluentbitvm:~/code/fluent-bit/9172_debug/bin$  valgrind ./fluent-bit -c fluent.conf 
==276354== Memcheck, a memory error detector
==276354== Copyright (C) 2002-2024, and GNU GPL'd, by Julian Seward et al.
==276354== Using Valgrind-3.23.0 and LibVEX; rerun with -h for copyright info
==276354== Command: ./fluent-bit -c fluent.conf
==276354== 
Fluent Bit v3.1.5
* Copyright (C) 2015-2024 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

______ _                  _    ______ _ _           _____  __  
|  ___| |                | |   | ___ (_) |         |____ |/  | 
| |_  | |_   _  ___ _ __ | |_  | |_/ /_| |_  __   __   / /`| | 
|  _| | | | | |/ _ \ '_ \| __| | ___ \ | __| \ \ / /   \ \ | | 
| |   | | |_| |  __/ | | | |_  | |_/ / | |_   \ V /.___/ /_| |_
\_|   |_|\__,_|\___|_| |_|\__| \____/|_|\__|   \_/ \____(_)___/

[2024/08/08 06:56:35] [ info] [fluent bit] version=3.1.5, commit=52ec818316, pid=276354
[2024/08/08 06:56:35] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/08 06:56:35] [ info] [cmetrics] version=0.9.3
[2024/08/08 06:56:35] [ info] [ctraces ] version=0.5.3
[2024/08/08 06:56:35] [ info] [input:tail:input_tail_logs] initializing
[2024/08/08 06:56:35] [ info] [input:tail:input_tail_logs] storage_strategy='memory' (memory only)
[2024/08/08 06:56:36] [ info] [input:tail:input_tail_logs] db: delete unmonitored stale inodes from the database: count=0
[2024/08/08 06:56:36] [ info] [sp] stream processor started
[2024/08/08 06:56:36] [ info] [output:stdout:stdout.0] worker #0 started
[2024/08/08 06:56:36] [ info] [input:tail:input_tail_logs] inotify_fs_add(): inode=548098 watch_fd=1 name=fake.log
^C[2024/08/08 06:56:42] [engine] caught signal (SIGINT)
[2024/08/08 06:56:42] [ warn] [engine] service will shutdown in max 5 seconds
[2024/08/08 06:56:42] [ info] [input] pausing input_tail_logs
[2024/08/08 06:56:43] [ info] [engine] service has stopped (0 pending tasks)
[2024/08/08 06:56:43] [ info] [input] pausing input_tail_logs
[2024/08/08 06:56:43] [ info] [output:stdout:stdout.0] thread worker #0 stopping...
[2024/08/08 06:56:43] [ info] [output:stdout:stdout.0] thread worker #0 stopped
[2024/08/08 06:56:43] [ info] [input:tail:input_tail_logs] inotify_fs_remove(): inode=548098 watch_fd=1
==276354== 
==276354== HEAP SUMMARY:
==276354==     in use at exit: 0 bytes in 0 blocks
==276354==   total heap usage: 2,177 allocs, 2,177 frees, 5,741,862 bytes allocated
==276354== 
==276354== All heap blocks were freed -- no leaks are possible
==276354== 
==276354== For lists of detected and suppressed errors, rerun with: -s
==276354== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)

```
</details> 
If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**
<!-- Docs can be edited at https://github.com/fluent/fluent-bit-docs -->
- [ ] Documentation required for this feature

<!--  Doc PR (not required but highly recommended) -->

**Backporting**
<!--
PRs targeting the default master branch will go into the next major release usually.
If this PR should be backported to the current or earlier releases then please submit a PR for that particular branch.
-->
- [ ] Backport to latest stable release.

<!--  Other release PR (not required but highly recommended for quick turnaround) -->
----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

rightblank commented:
Hi, @cosmo0920, would you please help to review this PR? Thanks!

rightblank commented:
Hi, @cosmo0920, I just updated the doc and the warning message in the code change, would you please help to trigger the CI again? Thanks!

cosmo0920 commented:
> Hi, @cosmo0920, I just updated the doc and the warning message in the code change, would you please help to trigger the CI again? Thanks!

CI, is already triggered automatically.

rightblank commented:
@cosmo0920, How does the change look to you? Could you please give an approval?

edsiper commented on README.md:
```diff
@@ -19,7 +19,7 @@ Fluent Bit allows to collect log events or metrics from different sources, proce
 Fluent Bit comes with full SQL [Stream Processing](https://docs.fluentbit.io/manual/stream-processing/introduction) capabilities: data manipulation and analytics using SQL queries.
 
-Fluent Bit runs on x86_64, x86, arm32v7, and arm64v8 architectures.
+Fluent Bit runs on x86_64, x86, s390x, arm32v7, and arm64v8 architectures.
```
we don't support s390x, we don't have any infrastructure or CI runners for it.

rightblank replied:
I have removed this change!


> we don't support s390x, we don't have any infrastructure or CI runners for it.

@edsiper , The IBM LinuxONE Community Cloud provides free access to developers, registering an account and creating a  s390x VM costs less than 30 minutes. https://community.ibm.com/zsystems/l1cc/

rightblank replied:
> we don't support s390x, we don't have any infrastructure or CI runners for it.

Hi, @edsiper, would you please clarify what do you mean by fluent bit don't support s390x? Do you mean 
1) the change in the readme, or,
2) you won't support any change made for s390x and thus not merge this PR?

- If for the firts reason, I have updated the readme already,
- If for the second reason, IBM has provided free s390x infrastructures for OSS communities here  https://community.ibm.com/zsystems/l1cc/, we can work together to add Z nodes into your CI pipelines.

- The last thing I want to mention is that the code change in this PR won't break anything outside of s390x platform, it just allows the user to build fluent bit with system luajit lib, it should be OK to be merged, what do you think?

rightblank replied:
I forgot to mention that I have demoed this change to @pwhelan and Adriana in the [community meeting](https://www.meetup.com/fluent-community-meeting/events/zdrqltygcmbhb/) last week and it works perfectly on Z.
so would you guys please help to merge this PR? @edsiper @pwhelan @cosmo0920

rightblank commented:
Hi, @edsiper @cosmo0920, would you please help to review again? 

- I added the test case output of flb-rt-filter_lua below.
- please let me know if the doc in I updated [here](https://github.com/fluent/fluent-bit/pull/9172/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R107) is OK?

<details><summary> test case output of flb-rt-filter_lua</summary>

```
linux1@fluentbitvm:~/code/fluent-bit/build/bin$  uname -a
Linux fluentbitvm 5.15.0-117-generic #127-Ubuntu SMP Fri Jul 5 20:13:04 UTC 2024 s390x s390x s390x GNU/Linux
linux1@fluentbitvm:~/code/fluent-bit/build/bin$  ./flb-rt-filter_lua -vvvvv
Test hello_world:
  filter_lua.c:427: Check ret == 0... ok
  filter_lua.c:430: Check filter_ffd >= 0... ok
  filter_lua.c:440: Check in_ffd >= 0... ok
  filter_lua.c:444: Check out_ffd >= 0... ok
[2024/08/13 06:30:55] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538004
[2024/08/13 06:30:55] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:30:55] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:30:55] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:30:55] [ info] [input:dummy:dummy.0] initializing
[2024/08/13 06:30:55] [ info] [input:dummy:dummy.0] storage_strategy='memory' (memory only)
[2024/08/13 06:30:55] [ info] [sp] stream processor started
  filter_lua.c:450: Check ret==0... ok
[2024/08/13 06:30:55] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:30:55] [ info] [input] pausing dummy.0
[2024/08/13 06:30:55] [ info] [output:stdout:stdout.0] worker #0 started
[2024/08/13 06:30:56] [ info] [engine] service has stopped (0 pending tasks)
[2024/08/13 06:30:56] [ info] [input] pausing dummy.0
[2024/08/13 06:30:56] [ info] [output:stdout:stdout.0] thread worker #0 stopping...
[2024/08/13 06:30:56] [ info] [output:stdout:stdout.0] thread worker #0 stopped
  SUCCESS: All conditions have passed.

Test append_tag:
  filter_lua.c:368: Check ret == 0... ok
  filter_lua.c:371: Check filter_ffd >= 0... ok
  filter_lua.c:381: Check in_ffd >= 0... ok
  filter_lua.c:385: Check out_ffd >= 0... ok
[2024/08/13 06:30:56] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538008
[2024/08/13 06:30:56] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:30:56] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:30:56] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:30:56] [ info] [input:lib:lib.0] initializing
[2024/08/13 06:30:56] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/13 06:30:56] [ info] [sp] stream processor started
  filter_lua.c:392: Check ret==0... ok
  filter_lua.c:397: Check result != NULL... ok
[2024/08/13 06:30:57] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:30:58] [ info] [engine] service has stopped (0 pending tasks)
  SUCCESS: All conditions have passed.

Test type_int_key:
  filter_lua.c:204: Check ret == 0... ok
  filter_lua.c:207: Check filter_ffd >= 0... ok
  filter_lua.c:218: Check in_ffd >= 0... ok
  filter_lua.c:222: Check out_ffd >= 0... ok
[2024/08/13 06:30:58] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538011
[2024/08/13 06:30:58] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:30:58] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:30:58] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:30:58] [ info] [input:lib:lib.0] initializing
[2024/08/13 06:30:58] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/13 06:30:58] [ info] [sp] stream processor started
  filter_lua.c:229: Check ret==0... ok
  filter_lua.c:234: Check result == NULL... ok
  filter_lua.c:238: Check result != NULL... ok
[2024/08/13 06:30:59] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:31:00] [ info] [engine] service has stopped (0 pending tasks)
  SUCCESS: All conditions have passed.

Test type_int_key_multi:
  filter_lua.c:281: Check ret == 0... ok
  filter_lua.c:284: Check filter_ffd >= 0... ok
  filter_lua.c:295: Check in_ffd >= 0... ok
  filter_lua.c:299: Check out_ffd >= 0... ok
[2024/08/13 06:31:00] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538014
[2024/08/13 06:31:00] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:31:00] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:31:00] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:31:00] [ info] [input:lib:lib.0] initializing
[2024/08/13 06:31:00] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/13 06:31:00] [ info] [sp] stream processor started
  filter_lua.c:306: Check ret==0... ok
  filter_lua.c:313: Check result == NULL... ok
  filter_lua.c:317: Check result == NULL... ok
  filter_lua.c:323: Check result != NULL... ok
  filter_lua.c:327: Check result != NULL... ok
[2024/08/13 06:31:01] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:31:02] [ info] [engine] service has stopped (0 pending tasks)
  SUCCESS: All conditions have passed.

Test type_array_key:
  filter_lua.c:489: Check ret == 0... ok
  filter_lua.c:492: Check filter_ffd >= 0... ok
  filter_lua.c:503: Check in_ffd >= 0... ok
  filter_lua.c:507: Check out_ffd >= 0... ok
[2024/08/13 06:31:02] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538018
[2024/08/13 06:31:02] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:31:02] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:31:02] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:31:02] [ info] [input:lib:lib.0] initializing
[2024/08/13 06:31:02] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
  filter_lua.c:514: Check ret==0... ok
[2024/08/13 06:31:02] [ info] [sp] stream processor started
  filter_lua.c:519: Check result != NULL... ok
  filter_lua.c:523: Check result != NULL... ok
[2024/08/13 06:31:03] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:31:04] [ info] [engine] service has stopped (0 pending tasks)
  SUCCESS: All conditions have passed.

Test array_contains_null:
  filter_lua.c:566: Check ret == 0... ok
  filter_lua.c:569: Check filter_ffd >= 0... ok
  filter_lua.c:579: Check in_ffd >= 0... ok
  filter_lua.c:583: Check out_ffd >= 0... ok
[2024/08/13 06:31:04] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538021
[2024/08/13 06:31:04] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:31:04] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:31:04] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:31:04] [ info] [input:lib:lib.0] initializing
[2024/08/13 06:31:04] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
  filter_lua.c:590: Check ret==0... ok
[2024/08/13 06:31:04] [ info] [sp] stream processor started
  filter_lua.c:595: Check result != NULL... ok
  filter_lua.c:599: Check result != NULL... ok
[2024/08/13 06:31:05] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:31:06] [ info] [engine] service has stopped (0 pending tasks)
  SUCCESS: All conditions have passed.

Test drop_all_records:
  filter_lua.c:639: Check ret == 0... ok
  filter_lua.c:642: Check filter_ffd >= 0... ok
  filter_lua.c:652: Check in_ffd >= 0... ok
  filter_lua.c:656: Check out_ffd >= 0... ok
[2024/08/13 06:31:06] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538024
[2024/08/13 06:31:06] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:31:06] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:31:06] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:31:06] [ info] [input:lib:lib.0] initializing
[2024/08/13 06:31:06] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/13 06:31:06] [ info] [sp] stream processor started
  filter_lua.c:662: Check ret==0... ok
  filter_lua.c:668: Check ret == 0... ok
[2024/08/13 06:31:07] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:31:08] [ info] [engine] service has stopped (0 pending tasks)
  SUCCESS: All conditions have passed.

Test enable_flb_null:
  filter_lua.c:708: Check ret == 0... ok
  filter_lua.c:711: Check filter_ffd >= 0... ok
  filter_lua.c:722: Check in_ffd >= 0... ok
  filter_lua.c:726: Check out_ffd >= 0... ok
[2024/08/13 06:31:08] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538027
[2024/08/13 06:31:08] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:31:08] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:31:08] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:31:08] [ info] [input:lib:lib.0] initializing
[2024/08/13 06:31:08] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/13 06:31:08] [ info] [flb_lua_enable_flb_null] set flb_null
[2024/08/13 06:31:08] [ info] [sp] stream processor started
  filter_lua.c:733: Check ret==0... ok
  filter_lua.c:738: Check result != NULL... ok
[2024/08/13 06:31:09] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:31:10] [ info] [engine] service has stopped (0 pending tasks)
  SUCCESS: All conditions have passed.

Test split_record:
  filter_lua.c:785: Check ret == 0... ok
  filter_lua.c:788: Check filter_ffd >= 0... ok
  filter_lua.c:798: Check in_ffd >= 0... ok
  filter_lua.c:802: Check out_ffd >= 0... ok
[2024/08/13 06:31:10] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538030
[2024/08/13 06:31:10] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:31:10] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:31:10] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:31:10] [ info] [input:lib:lib.0] initializing
[2024/08/13 06:31:10] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/13 06:31:10] [ info] [sp] stream processor started
  filter_lua.c:809: Check ret==0... ok
[2024/08/13 06:31:12] [ warn] [timeout] elapsed_time: 2001
  filter_lua.c:813: Check !strcmp(outbuf, expected)... ok
[2024/08/13 06:31:12] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:31:13] [ info] [engine] service has stopped (0 pending tasks)
  SUCCESS: All conditions have passed.

Test empty_array:
  filter_lua.c:856: Check ret == 0... ok
  filter_lua.c:859: Check filter_ffd >= 0... ok
  filter_lua.c:869: Check in_ffd >= 0... ok
  filter_lua.c:873: Check out_ffd >= 0... ok
[2024/08/13 06:31:13] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538033
[2024/08/13 06:31:13] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:31:13] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:31:13] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:31:13] [ info] [input:lib:lib.0] initializing
[2024/08/13 06:31:13] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/13 06:31:13] [ info] [sp] stream processor started
  filter_lua.c:880: Check ret==0... ok
[2024/08/13 06:31:15] [ warn] [timeout] elapsed_time: 2002
  filter_lua.c:884: Check !strcmp(outbuf, expected)... ok
[2024/08/13 06:31:15] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:31:16] [ info] [engine] service has stopped (0 pending tasks)
  SUCCESS: All conditions have passed.

Test invalid_metatable:
  filter_lua.c:927: Check ret == 0... ok
  filter_lua.c:930: Check filter_ffd >= 0... ok
  filter_lua.c:940: Check in_ffd >= 0... ok
  filter_lua.c:944: Check out_ffd >= 0... ok
[2024/08/13 06:31:16] [ info] [fluent bit] version=3.1.5, commit=a0cba84168, pid=538036
[2024/08/13 06:31:16] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/13 06:31:16] [ info] [cmetrics] version=0.9.3
[2024/08/13 06:31:16] [ info] [ctraces ] version=0.5.3
[2024/08/13 06:31:16] [ info] [input:lib:lib.0] initializing
[2024/08/13 06:31:16] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
  filter_lua.c:950: Check ret==0... ok
  filter_lua.c:953: Check ret != -1... ok
[2024/08/13 06:31:16] [ info] [sp] stream processor started
  filter_lua.c:82: Check data != NULL... ok
  filter_lua.c:959: Check ret > 0... ok
[2024/08/13 06:31:17] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/13 06:31:18] [ info] [engine] service has stopped (0 pending tasks)
  SUCCESS: All conditions have passed.

Summary:
  Count of all unit tests:       11
  Count of run unit tests:       11
  Count of failed unit tests:     0
  Count of skipped unit tests:    0
SUCCESS: All unit tests have passed.
```

</details>

rightblank commented:
@leonardo-albertovich, would you please try this PR on your s390x vm and give it a try?
LuaJit support on s390x is not merged upstream but maintained in https://github.com/LuaJIT/LuaJIT/pull/631, please clone that repo and run `make && make install`

edsiper commented:
@rightblank my last concern to fix is the word `supported`. While Fluent Bit can run on s390x we cannot commit time to troubleshoot issues that are reported in that area, so my next question are:

- is IBM willing to provide time to troubleshoot and contribute fixes if needed ?
- would you be ok to change the word `supported` with something like `it runs` ?

rightblank commented:
> @rightblank my last concern to fix is the word `supported`. While Fluent Bit can run on s390x we cannot commit time to troubleshoot issues that are reported in that area, so my next question are:
> 
> * is IBM willing to provide time to troubleshoot and contribute fixes if needed ?
> * would you be ok to change the word `supported` with something like `it runs` ?

@edsiper, I just updated the README to replace `supported` to `it runs`, is it OK for you to merge this PR?

rightblank commented:
And if fluent bit uses [openresty/luajit2](https://github.com/openresty/luajit2), fluent bit can even support embed LuaJIT for s390x.

rightblank commented:
<details><summary>Lua Test case output with openresty/luajit2</summary>

```
linux1@fluentbitvm:~/code/fluent-bit/build/bin$  ./flb-it-lua 
Test lua_is_valid_func...                       [ OK ]
Test lua_pushtimetable...                       [ OK ]
Test lua_pushmsgpack...                         [ OK ]
Test lua_pushmpack...                           [ OK ]
Test lua_tomsgpack...                           [ OK ]
Test lua_tompack...                             [ OK ]
Test lua_arraylength...                         [ OK ]
Test lua_arraylength_with_index...              [ OK ]
Test lua_arraylength_for_array_contains_nil...  [ OK ]
SUCCESS: All unit tests have passed.
linux1@fluentbitvm:~/code/fluent-bit/build/bin$   ./flb-rt-filter_lua
Test hello_world...                             [2024/08/15 14:33:41] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702312
[2024/08/15 14:33:41] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:33:41] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:33:41] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:33:41] [ info] [input:dummy:dummy.0] initializing
[2024/08/15 14:33:41] [ info] [input:dummy:dummy.0] storage_strategy='memory' (memory only)
[2024/08/15 14:33:41] [ info] [output:stdout:stdout.0] worker #0 started
[2024/08/15 14:33:41] [ info] [sp] stream processor started
[2024/08/15 14:33:41] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:33:41] [ info] [input] pausing dummy.0
[2024/08/15 14:33:42] [ info] [engine] service has stopped (0 pending tasks)
[2024/08/15 14:33:42] [ info] [input] pausing dummy.0
[2024/08/15 14:33:42] [ info] [output:stdout:stdout.0] thread worker #0 stopping...
[2024/08/15 14:33:42] [ info] [output:stdout:stdout.0] thread worker #0 stopped
[ OK ]
Test append_tag...                              [2024/08/15 14:33:42] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702316
[2024/08/15 14:33:42] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:33:42] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:33:42] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:33:42] [ info] [input:lib:lib.0] initializing
[2024/08/15 14:33:42] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/15 14:33:42] [ info] [sp] stream processor started
[2024/08/15 14:33:43] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:33:44] [ info] [engine] service has stopped (0 pending tasks)
[ OK ]
Test type_int_key...                            [2024/08/15 14:33:44] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702319
[2024/08/15 14:33:44] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:33:44] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:33:44] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:33:44] [ info] [input:lib:lib.0] initializing
[2024/08/15 14:33:44] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/15 14:33:44] [ info] [sp] stream processor started
[2024/08/15 14:33:45] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:33:46] [ info] [engine] service has stopped (0 pending tasks)
[ OK ]
Test type_int_key_multi...                      [2024/08/15 14:33:46] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702322
[2024/08/15 14:33:46] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:33:46] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:33:46] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:33:46] [ info] [input:lib:lib.0] initializing
[2024/08/15 14:33:46] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/15 14:33:46] [ info] [sp] stream processor started
[2024/08/15 14:33:47] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:33:48] [ info] [engine] service has stopped (0 pending tasks)
[ OK ]
Test type_array_key...                          [2024/08/15 14:33:48] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702325
[2024/08/15 14:33:48] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:33:48] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:33:48] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:33:48] [ info] [input:lib:lib.0] initializing
[2024/08/15 14:33:48] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/15 14:33:48] [ info] [sp] stream processor started
[2024/08/15 14:33:49] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:33:50] [ info] [engine] service has stopped (0 pending tasks)
[ OK ]
Test array_contains_null...                     [2024/08/15 14:33:50] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702328
[2024/08/15 14:33:50] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:33:50] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:33:50] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:33:50] [ info] [input:lib:lib.0] initializing
[2024/08/15 14:33:50] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/15 14:33:50] [ info] [sp] stream processor started
[2024/08/15 14:33:51] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:33:52] [ info] [engine] service has stopped (0 pending tasks)
[ OK ]
Test drop_all_records...                        [2024/08/15 14:33:52] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702331
[2024/08/15 14:33:52] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:33:52] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:33:52] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:33:52] [ info] [input:lib:lib.0] initializing
[2024/08/15 14:33:52] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/15 14:33:52] [ info] [sp] stream processor started
[2024/08/15 14:33:53] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:33:54] [ info] [engine] service has stopped (0 pending tasks)
[ OK ]
Test enable_flb_null...                         [2024/08/15 14:33:54] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702334
[2024/08/15 14:33:54] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:33:54] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:33:54] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:33:54] [ info] [input:lib:lib.0] initializing
[2024/08/15 14:33:54] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/15 14:33:54] [ info] [flb_lua_enable_flb_null] set flb_null
[2024/08/15 14:33:54] [ info] [sp] stream processor started
[2024/08/15 14:33:55] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:33:56] [ info] [engine] service has stopped (0 pending tasks)
[ OK ]
Test split_record...                            [2024/08/15 14:33:56] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702337
[2024/08/15 14:33:56] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:33:56] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:33:56] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:33:56] [ info] [input:lib:lib.0] initializing
[2024/08/15 14:33:56] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/15 14:33:56] [ info] [sp] stream processor started
[2024/08/15 14:33:58] [ warn] [timeout] elapsed_time: 2002
[2024/08/15 14:33:58] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:33:59] [ info] [engine] service has stopped (0 pending tasks)
[ OK ]
Test empty_array...                             [2024/08/15 14:33:59] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702340
[2024/08/15 14:33:59] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:33:59] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:33:59] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:33:59] [ info] [input:lib:lib.0] initializing
[2024/08/15 14:33:59] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/15 14:33:59] [ info] [sp] stream processor started
[2024/08/15 14:34:01] [ warn] [timeout] elapsed_time: 2002
[2024/08/15 14:34:01] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:34:02] [ info] [engine] service has stopped (0 pending tasks)
[ OK ]
Test invalid_metatable...                       [2024/08/15 14:34:02] [ info] [fluent bit] version=3.1.6, commit=3ae9a75a39, pid=702343
[2024/08/15 14:34:02] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/08/15 14:34:02] [ info] [cmetrics] version=0.9.4
[2024/08/15 14:34:02] [ info] [ctraces ] version=0.5.5
[2024/08/15 14:34:02] [ info] [input:lib:lib.0] initializing
[2024/08/15 14:34:02] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2024/08/15 14:34:02] [ info] [sp] stream processor started
[2024/08/15 14:34:03] [ warn] [engine] service will shutdown in max 1 seconds
[2024/08/15 14:34:04] [ info] [engine] service has stopped (0 pending tasks)
[ OK ]
SUCCESS: All unit tests have passed.

```
</details>

edsiper commented:
cc: @tarruda

tarruda reviewed (approved):
Not familiar with s390x, but LGTM

rightblank commented:
Hi, @edsiper, would you please merge this PR?
